core/frontend/src/tile/GltfReader.ts

```diff
@@ -2172,7 +2172,17 @@ export abstract class GltfReader {
     const offset = bv.byteOffset ?? 0;
     buf = buf.subarray(offset, offset + bv.byteLength);
 
-    const mesh = await loader.parse(buf, { modules: { draco3d } }); // NB: `options` argument declared optional but will produce exception if not supplied. Regardless, we are specifying our own bundled draco3d module to avoid CDN requests.
+    const mesh = await loader.parse(buf, {
+      draco: {
+        decoderType: "wasm",
+      },
+      modules: {
+        "draco_wasm_wrapper.js": `${IModelApp.publicPath}draco/draco_wasm_wrapper.js`,
```

Comment:
does this actually work? is this not an issue or did i incorrect configure things before? https://github.com/visgl/loaders.gl/issues/3185

we dont need to resolve the module ourselves?

Reply:
the issue is the loaders.gl code doesn't actually want us to resolve the module ourselves. When you passed in the arrayBuffer representing the module, that gets fed into the [libraryUrl](https://github.com/visgl/loaders.gl/blob/d74df445180408fd772567ca6a6bbb8d42aa50be/modules/worker-utils/src/lib/library-utils/library-utils.ts#L40) variable in their loadLibrary() method. Then javascript, expecting a primitive string value, threw a `Cannot convert object to primitive value`. So libraryUrl does, actually, mean a url to the relevant assets, so I changed it to pass in the url to fetch the asset, and laoders.gl uses that url to call `loadLibraryFromFile`, the RHS of the `||` operator in the link above

Reply:
ok maybe respond to my comment on the issue then lol; nice fine